src/pyvesync/vesyncbulb.py

```diff
@@ -391,3 +497,338 @@ def set_color_temp(self, color_temp: int) -> bool:
+            logger.debug('Error getting %s config info', self.device_name)
+            logger.debug('  return code - %d with message %s',
+                         r.get('code'), r.get('msg'))
+
+    def __build_config_dict(self, conf_dict: Dict[str, str]) -> None:
+        """Build configuration dict for Multicolor bulb."""
+        self.config['currentFirmVersion'] = (
+            conf_dict.get('currentFirmVersion', ''))
+        self.config['latestFirmVersion'] = (
+            conf_dict.get('latestFirmVersion', ''))
+        self.config['firmwareUrl'] = (
+            conf_dict.get('firmwareUrl', ''))
+        self.config['allowNotify'] = (
+            conf_dict.get('allowNotify', ''))
+        self.config['deviceImg'] = (
+            conf_dict.get('deviceImg', ''))
+        self.config['defaultDeviceImg'] = (
+            conf_dict.get('defaultDeviceImg', ''))
+        self.config['ownerShip'] = (
+            conf_dict.get('ownerShip', False))
+
+    def toggle(self, status) -> bool:
+        """Toggle multicolor bulb."""
+        body = helpers.req_body(self.manager, 'bypassV2')
+        if status == 'off':
+            status_bool = False
+        else:
+            status_bool = True
+        body['cid'] = self.cid
+        body['configModule'] = self.config_module
+        body['payload'] = {
+            'method': 'setSwitch',
+            'source': 'APP',
+            'data': {
+                'id': 0,
+                'enabled': status_bool,
+                }
+            }
+        r, _ = helpers.call_api(
+            '/cloud/v2/deviceManaged/bypassV2',
+            'post',
+            headers=helpers.req_header_bypass(),
+            json_object=body)
+        if helpers.code_check(r):
+            self.device_status = status
+            return True
+        logger.debug('%s offline', self.device_name)
+        self.device_status = 'off'
+        self.connection_status = 'offline'
+        return False
+
+    def set_brightness(self, brightness: int) -> bool:
+        """Set brightness of multicolor bulb."""
+        return self.turn_on(brightness=brightness)
+
+    def set_color_value(self, color_value: int) -> bool:
+        """Set color value of multicolor bulb."""
+        return self.turn_on(color_value=color_value)
+
+    def set_color_temp(self, color_temp: int) -> bool:
+        """Set White Temperature of Bulb in pct (0 - 100)."""
+        return self.turn_on(color_temp=color_temp)
+
+    def set_color_saturation(self, color_saturation: int) -> bool:
+        """Set Color Saturation of Bulb in pct (1 - 100)."""
+        return self.turn_on(color_saturation=color_saturation)
+
+    def set_color_hue(self, color_hue: float) -> bool:
+        """Set Color Hue of Bulb (0 - 360)."""
+        return self.turn_on(color_hue=color_hue)
+
+    def set_color_mode(self, color_mode: str) -> bool:
+        """Set Color Mode of Bulb (white / hsv)."""
+        return self.turn_on(color_mode=color_mode)
+
+    def turn_on(self,
```

Comment:
Make this a `set_status()` method that abstracts the API call. The logic for each of the brightness, color_temp, saturation, etc. should have its own method.

Reply:
set_status() method done, I'll mark this as resolved, pardon me if not.